Homework/Tai_Vo/lession01/TaiVo283MoveZeroes.java

```diff
@@ -0,0 +1,48 @@
+package lession01;
+
+import java.util.Arrays;
+
+/**
+ * @author taivt
+ * @link https://leetcode.com/problems/move-zeroes/
+ * @since 2022/03/14 21:13:06
+ */
+public class TaiVo283MoveZeroes {
+    public static void main(String[] args) {
+        int[] nums = {0, 1, 0, 3, 12};
+        new Solution().moveZeroes(nums);
+        System.out.println(Arrays.toString(nums));
+
+        int[] nums2 = {0};
+        new Solution().moveZeroes(nums2);
+        System.out.println(Arrays.toString(nums2));
+
+        int[] nums3 = {4, 2, 4, 0, 0, 3, 0, 5, 1, 0};
+        new Solution().moveZeroes(nums3);
+        System.out.println(Arrays.toString(nums3));
+    }
+
+    // Space Complexity: O(1)
+    // Time Complexity: O(n)
+    private static class Solution {
+        public void moveZeroes(int[] nums) {
+            int cur = 0;
+            int nextNonZeroIdx = 1;
+
+            while (nextNonZeroIdx < nums.length && cur < nums.length) {
```

Comment:
will it maintain the relative order of the non-zero elements?

Reply:
Yes, it will. Because it will swap the current number and the next non-zero number, and the position of the current number is always less than the position of the next non-zero number. Therefore it will maintain the relative order of non-zero elements.